lib/Dialect/TritonGPU/Transforms/Pipeliner/MatmulLoopPipeline.cpp

```diff
@@ -334,9 +334,14 @@ getSharedEncoding(Operation *loadOp, bool isMMAV3Shared) {
   } else {
     order = blockedOrder;
   }
-  if (isMMAV3Shared) {
-    return ttg::SharedEncodingAttr::get(ty.getContext(), ty.getShape(), order,
-                                        ctaLayout, ty.getElementType());
+
+  auto mmav3Enc = ttg::SharedEncodingAttr::get(
+      ty.getContext(), ty.getShape(), order, ctaLayout, ty.getElementType());
+
+  if (isMMAV3Shared && isTMALoad) {
+    // For TMA, the encoding compatible with it takes precedence over local
+    // alloc created for the MMA operand.
+    return mmav3Enc;
   }
```

Comment:
shouldn't the condition just be `if (isTMALoad)` here? Also I wonder if we can just remove `isMMAV3Shared` as it only makes sense when it feeds into a an AllocOp anyway

Reply:
ah right, I thought `isTMALoad = true` doesn't always imply `isMMAV3Shared` because TMA can be used with `mma.sync`. But I remember you said that `isMMAV3Shared` is supposed to be true in that case as well. 

> Also I wonder if we can just remove isMMAV3Shared as it only makes sense when it feeds into a an AllocOp anyway

With the change you suggested, the only use of `isMMAV3Shared` in this function is the case for mullti-buffering MMAv2 operands. In this case, `AccelerateMatmul` does not create local alloc for operands, so I thought we still need this? 

```
  if (isMMAV3Shared) {
    // When there is no local alloc but MMAv3 encoding is requested - it is for
    // MMAv2
    return mmav3Enc;
  }
```

Reply:
> But I remember you said that isMMAV3Shared is supposed to be true for MMAv2 as well.

Only when the load is from TMA.



> With the change you suggested, the only use of `isMMAV3Shared` in this function is the case for mullti-buffering MMAv2 operands. In this case, `AccelerateMatmul` does not create local alloc for operands, so I thought we still need this?
> 
> ```
>   if (isMMAV3Shared) {
>     // When there is no local alloc but MMAv3 encoding is requested - it is for
>     // MMAv2
>     return mmav3Enc;
>   }
> ```

We only want that when the load is TMA so I think this is already handled by the rest of the code

Reply:
I see, I didn't realize that there is another function `getSharedEncIfAllUsersAreDotEnc` that creates SMEM encoding for sm80. 

Removed `isMMAV3Shared`.